pg_nearest_city/scripts/voronoi_generator.py

```diff
@@ -35,24 +84,67 @@ class Config:
     db_host: str = os.environ.get("PGNEAREST_DB_HOST", "localhost")
     db_port: int = int(os.environ.get("PGNEAREST_DB_PORT", "5432"))
 
-    # Data sources
-    geonames_url: str = "http://download.geonames.org/export/dump/cities1000.zip"
-    _zip_path: str = ""
+    # Cache configuration
+    cache_dir: Path = Path("./cache")
+    cur_dt: datetime = datetime.now()
 
     # Output configuration
-    output_dir: Path = Path("/data/output")  # Default output directory
+    output_dir: Path = Path("/data/output")
+    cache_files: bool = True
     compress_output: bool = True
 
     # Processing options
     country_filter: Optional[str] = None  # Optional filter for testing (e.g., "IT")
 
+    # Data sources
+    country_boundaries: URLConfig = field(init=False)
+    geonames: URLConfig = field(init=False)
+
+    def __post_init__(self) -> None:
+        """Creates URLConfig objects for Config."""
+        self.geonames: URLConfig = URLConfig(
+            domain="download.geonames.org",
+            path="export/dump/cities500.zip",
+            zip_name="cities500.zip",
+        )
+        self.geonames_old: URLConfig = URLConfig(
```

Comment:
This and the others are being temporarily maintained while the desired data sources are decided in #33 .